### PR TITLE
feat: add nepali date value prop

### PR DESCRIPTION
**Pull request by sanamhub**
Just get NepaliDate value with `.Value`

Cannot create `NepaliDate` prop, due to the same class name.

### CODE REVIEW

**Comment by RajuPrasai** [comment]
What's wrong with .ToString()?
In future, if we have to return the string in the desired format, then ToString() would be better which is also done in DateTime struct.
And conventionally speaking. We want to get the value as string. So ToString() seems reasonable.

**Comment by sanamhub** [comment]
> What's wrong with .ToString()? In future, if we have to return the string in the desired format, then ToString() would be better which is also done in DateTime struct. And conventionally speaking. We want to get the value as string. So ToString() seems reasonable.

Generally, `DateTime` don't require to return string and they are just using ToString with some overloads to format DateTime to certain format. Our case seems different, we/ide generally show/search for a peoperty where NepaliDate (Value for our case) property could seems suitable for easy and convinent accessing of the class properties. ToString() seems kind of extension method and looks forcefully converting to string 😃  (though we have our own impl)

I mean we can have both prop and extension, ToString() can be used for formatting with overloads like in .NET

**Comment by RajuPrasai** [comment]
> > What's wrong with .ToString()? In future, if we have to return the string in the desired format, then ToString() would be better which is also done in DateTime struct. And conventionally speaking. We want to get the value as string. So ToString() seems reasonable.
> 
> Generally, `DateTime` don't require to return string and they are just using ToString with some overloads to format DateTime to certain format. Our case seems different, we/ide generally show/search for a peoperty where NepaliDate (Value for our case) property could seems suitable for easy and convinent accessing of the class properties. ToString() seems kind of extension method and looks forcefully converting to string 😃 (though we have our own impl)
> 
> I mean we can have both prop and extension, ToString() can be used for formatting with overloads like in .NET

Though, I appreciate your input with the changes. Here is my thought..

Firstly, `NepaliDate` also doesn't requires to be returned as a `string`. We can create a custom model binder that can handle the conversion between the `struct` and the input data provided from the view.
Secondly, conventionally Value of a `struct` seems to be the `struct` itself and `ToString()` method is a standard method in the .NET to represent the `struct` value as a string.

**Comment by sanamhub** [comment]
Thank you for your reply and suggestions. While I appreciate the use of custom model binders to handle the conversion between the NepaliDate struct and input data from the view, I believe that providing a NepaliDate property would enhance the usability and flexibility of the NepDate library.

By implementing a NepaliDate property, users of the library could access the NepaliDate value in a clear and intuitive manner. This approach aligns with the principles of object-oriented design, which emphasize encapsulating data and behavior within classes.

Regarding the use of the ToString() method, while it is a standard method in the .NET, its use for struct values can be limiting in terms of customization and flexibility. By providing both a NepaliDate property and the ToString() method with overloads, we can provide users with the ability to format the NepaliDate value in a way that meets their specific needs.

In summary, I suggest implementing a NepaliDate property in addition to the ToString() method with overloads. This approach aligns with object-oriented design principles and provides users with greater usability and flexibility. Thank you for considering my input.
